src/signal/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+//! signal implementation for Redox, following http://pubs.opengroup.org/onlinepubs/7908799/xsh/signal.h.html
+
+#![no_std]
+
+extern crate platform;
+
+use platform::types::*;
+
+#[cfg(target_os = "linux")]
+#[repr(C)]
+pub struct sigset_t {
+    pub bits: [u64; 16],
+}
+
+#[cfg(target_os = "redox")]
+#[repr(C)]
+pub struct sigset_t {
+    pub bits: [u64; 2],
+}
+
+mod inner {
+    use super::sigset_t;
+
+    #[repr(C)]
+    pub struct sigaction {
+        pub sa_handler: extern "C" fn(usize),
+        pub sa_mask: sigset_t,
+        pub sa_flags: usize,
+    }
+}
+
+// RedoxOS Signals
+
+#[cfg(target_os = "redox")]
```

Comment:
Look at fcntl for a better way to do this

Reply:
A separate mod for Redox and Linux

Reply:
AFAIK the constants don't get exported into the header with this though. I don't see the `O_RDONLY` constant in `fcntl.h`.

Reply:
I agree that using a sys crate is **much** more readable though. Is there a way to implement these like the `fcntl` crate and export the constants. Or is the fact that the constants are not exported a bug in `cbindgen`?

Reply:
I will look into it